### DISCUSSION TITLE
希望umi团队进一步提供有关umi插件单元测试的指引

### DISCUSSION THREAD

**Discussion by aspirantzhang** [umijs/umi#6740] [Q&A]
已经有很多issue反馈，umi/plugins在单元测试中的各类问题，包括useRequest/locale/connect/dispatch等，很多目前没有太完善的处理方法。我本人希望umi团队能够重视这些问题，提供完善处理指引。
注：单纯添加`moduleNameMapper`不能解决问题。
@sorrycc @ycjcl868 @chenshuai2144 @xiaohuoni @afc163

issue集合：
【plugin-request】
- @aspirantzhang #6484

【plugin-locale】
- @khalibloo #4358
- @tejparekh #6528
- @six-666 #4598
- @LingzhiLiu #2156

【plugin-dva】
- @zhangxiangqiang #5138
- @lhongtao #6720
- https://stackoverflow.com/questions/62935492/umijs-unit-test-error-umi-connect-is-not-a-function

**Comment by ycjcl868** [comment]
`import from 'umi'` 后插件的模块映射是个问题，偶尔会伴随一些循环依赖 bug，方案的话，得在 `@umijs/test` 里内置 `moduleNameMapper`，自动识别插件里的 `addUmiExports` 的模块映射
